_posts/2025-06-27-this-week-in-rails.markdown

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "BacktraceCleaner gets first_clean_frame and first_clean_location"
+categories: news
+author: zzak
+og_image: assets/images/this-week-in-rails.png
+published: true
+date: 2025-06-27
+---
+
+
+Hi, it's [zzak](https://github.com/zzak). Let's explore this week's changes in the Rails codebase.
+
+[The Rails Foundation launched a new podcast: 'On Rails' with host Robby Russell.](https://onrails.buzzsprout.com)  
+In each episode, Robby talks with Rails devs and explores the real-world decisions that go into building, maintaining, and scaling Ruby on Rails applications. Episode one is already up with guest Rosa Gutiérrez.
+Listen at [onrails.buzzsprout.com](https://onrails.buzzsprout.com), or in your favorite podcast app.
+
+[Improve CurrentAttribute and ExecutionContext state management in test cases](https://github.com/rails/rails/pull/55247)  
+Previously these two global state would be entirely cleared out whenever calling into code that is wrapped by the Rails executor, typically Action Controller or Active Job helpers.
+Now re-entering the executor will properly save and restore that state.
+
+[Implement ActiveSupport::BacktraceCleaner#first_clean_frame](https://github.com/rails/rails/pull/55222)  
+The new method `ActiveSupport::BacktraceCleaner#first_clean_frame` returns the first clean frame of the caller's backtrace, or `nil`.
+Useful when you want to report the application-level frame where something happened as a string.
```

Comment:
Maybe this could be somehow rephrased. The "something happened as a string" sounds confusing